Library/Invoice.cs

```diff
@@ -586,6 +589,21 @@ internal void ReadXml(XmlTextReader reader, string nodeName)
 
                     case "address":
                         Address = new Address(reader);
+                        _referenceAddress = new Address()
```

Comment:
This seems pretty fragile, could you not clone this object?

Reply:
I didn't see a built-in `DeepCopy` method for this class, but I can do a bit more research to determine if there is a more robust option for cloning the Address object

Reply:
Added a `Address.Clone()` method, which uses `MemberwiseClone` under the hood. Less fragile?